python-checks/src/main/java/org/sonar/python/checks/hotspots/HardCodedCredentialsEntropyCheck.java

```diff
@@ -0,0 +1,214 @@
+  }
+
+  private void checkParameterList(SubscriptionContext subscriptionContext) {
+    var parameterList = (ParameterList) subscriptionContext.syntaxNode();
+    parameterList.nonTuple().stream()
+      .filter(parameter -> parameter.name() != null)
+      .filter(parameter -> parameter.defaultValue() != null)
+      .filter(parameter -> parameter.defaultValue().is(Tree.Kind.STRING_LITERAL))
+      .forEach(parameter -> {
+        var value = ((StringLiteral) parameter.defaultValue()).trimmedQuotesValue();
+        patternMatch(parameter.name(), parameter.defaultValue(), value, subscriptionContext);
+      });
+  }
+
+  private void checkDictionaryLiteral(SubscriptionContext subscriptionContext) {
+    var dictionaryLiteral = (DictionaryLiteral) subscriptionContext.syntaxNode();
+    dictionaryLiteral.elements().stream().filter(e -> e.is(Tree.Kind.KEY_VALUE_PAIR))
+      .map(KeyValuePair.class::cast)
+      .filter(keyValuePair -> keyValuePair.value().is(Tree.Kind.STRING_LITERAL))
+      .forEach(keyValuePair -> {
+        var value = ((StringLiteral) keyValuePair.value()).trimmedQuotesValue();
+        var key = keyValuePair.key();
+        if (!key.is(Tree.Kind.STRING_LITERAL)) {
+          return;
+        }
+        patternMatch(((StringLiteral) key).trimmedQuotesValue(), keyValuePair.value(), value, subscriptionContext);
+      });
+  }
+
+  private void checkRegularArgument(SubscriptionContext subscriptionContext) {
+    var regularArgument = (RegularArgument) subscriptionContext.syntaxNode();
+    var keywordArgument = regularArgument.keywordArgument();
+    if (keywordArgument == null) {
+      return;
+    }
+    var expression = regularArgument.expression();
+    if (!expression.is(Tree.Kind.STRING_LITERAL)) {
+      return;
+    }
+    var value = ((StringLiteral) expression).trimmedQuotesValue();
+    patternMatch(keywordArgument, regularArgument, value, subscriptionContext);
+  }
+
+  private void checkAnnotatedAssignment(SubscriptionContext subscriptionContext) {
+    var annotatedAssignment = (AnnotatedAssignment) subscriptionContext.syntaxNode();
+    var assignedValue = annotatedAssignment.assignedValue();
+    var assignedVariable = annotatedAssignment.variable();
+    if (assignedValue == null || !assignedValue.is(Tree.Kind.STRING_LITERAL) || !assignedVariable.is(Tree.Kind.NAME)) {
+      return;
+    }
+    var value = ((StringLiteral) assignedValue).trimmedQuotesValue();
+    patternMatch(((Name) assignedVariable).name(), assignedValue, value, subscriptionContext);
+  }
+
+  private void checkAssignment(SubscriptionContext subscriptionContext) {
+    var assignment = (AssignmentStatement) subscriptionContext.syntaxNode();
+    var assignedValue = assignment.assignedValue();
+    Map<Expression, Name> expressions = null;
+    if (assignedValue.is(Tree.Kind.TUPLE)) {
+      expressions = Expressions.getExpressionsFromRhs((assignment.assignedValue()))
```

Comment:
Do you need to use `assignment.assignedValue()` getter since you already have this value assigned to the `assignedValue` variable?
Also, there are redundant extra parenthesis